src/list/item/factory.js

```diff
@@ -535,15 +536,29 @@ export default /* @ngInject */ function ($q, $timeout) {
       } else {
         search = search.toLowerCase(); // ignore case
         const filteredItems = [];
-        let tmpMatchingItem;
-        angular.forEach(self.subItemsAdded, (item) => {
-          tmpMatchingItem = getMatchingItem(item, search);
 
-          if (tmpMatchingItem) {
-            filteredItems.push(tmpMatchingItem);
+        const promises = _.reduce(self.subItemsAdded, (results, item) => {
+          if (item.onLoad && !item.isLoaded) {
```

Comment:
This looks like a filter + map no ? 
```suggestion
         filter(self.subItemsAdded, item => item.onLoad && !item.isLoaded).map(item => item.loadSubItems())
```
I don't know if that's better in terms of readability/performances

Reply:
I said the exact same thing yesterday but my comment isn't the for some reason !

I agree fully with you :)